### DISCUSSION TITLE
bigger latency when drawing

### DISCUSSION THREAD

**Discussion by Wladefant** [personalizedrefrigerator/js-draw#37] [General]
It seems that after the new update there is a visible latency when drawing, especially when comparing side by side with samsung notes. Can that be or is it me hallucinating? did you do anything related to that in the latest update?

**Comment by personalizedrefrigerator** [comment]
Is "input stabilization" enabled? If so, try disabling it — stabilization can lead to a larger latency. 

If input stabilization is not enabled there might be a performance regression.

One possibility is that it's related to memory usage. Here are some other things to try to ensure it isn't related to the selected tool:
1. Restart Joplin
1. Create a new, empty note and attach a new drawing
2. Disable the grid background in the new drawing
3. Switch to the rounded tip pen
4. Compare latency again

Thank you for reporting this!

**Reply by Wladefant** [reply]
Yes, it was input stabilization

**Reply by Wladefant** [reply]
What does input stabilization do exactly. And it isn't default right?

**Reply by personalizedrefrigerator** [reply]
Input stabilization works by giving the tip of the stroke momentum (see [`InputStabilizer.ts`](https://github.com/personalizedrefrigerator/js-draw/blob/cf64a8bf9fccc54c87478ea6ed5da771346b62bf/packages/js-draw/src/tools/InputFilter/InputStabilizer.ts)) and updating it based on a spring force and friction. It should make smoother strokes, at the cost of latency.

It's the default on the middle pen.

**Reply by personalizedrefrigerator** [reply]
This could probably be improved by connecting the end of the smoothed/mapped stroke to the pen cursor (perhaps using the normal/non-smoothed drawing logic).

Current difficulty: stroke smoothing is currently set up as an input-event filter (there's currently no way to get the original input events, but this can be changed).

**Comment by manishgupta577** [comment]
hello ,can you please explain what changes i have to make in the settings object in my javascript code for tasks like changing the pen color,size,thickness etc .I tried according to the documentation but it could'nt help me

**Reply by personalizedrefrigerator** [reply]
See https://github.com/personalizedrefrigerator/js-draw/discussions/43
